`azurerm_machine_learning_workspace` - Add support for `serverless_compute` (#25660)
hashicorp/terraform-provider-azurerm: website/docs/r/machine_learning_workspace.html.markdown

---

> An `serverless_compute` block supports the following:

Reviewer: ```suggestion
A `serverless_compute` block supports the following:
```

> * `custom_subnet_id` - (Optional) The ID of an existing virtual network subnet in which serverless compute nodes should be deployed.

Reviewer: This is just called `subnet_id` in the schema
```suggestion
* `subnet_id` - (Optional) The ID of an existing Virtual Network Subnet in which the serverless compute nodes should be deployed to.
```

> * `no_public_ip_enabled` - (Optional) Is serverless compute nodes deployed in custom vNet would have no public IP addresses enabled for a workspace with private endpoint? Defaults to `false`.

Reviewer: ```suggestion
* `public_ip_enabled` - (Optional) Should serverless compute nodes deployed in a custom Virtual Network have public IP addresses enabled for a workspace with private endpoint? Defaults to `false`.
```

~> **Note:** Not supported to set `public_ip_enabled` to `false` without `subnet_id` when `public_network_access_enabled` is `false`.

> ~> **Note:** Not supported to update `public_ip_enabled` from `true` to `false` when `subnet_id` is null or empty

Reviewer 2: this needs to be updated

Author: delete the notes?

Author: to name the property `serverlessComputeNoPublicIP` to `public_ip_enabled`, value has been flipped. I add the constraint in the resource and also the doc.

> ~> **Note:** Not supported to set `public_ip_enabled` to `false` without `subnet_id` when `public_network_access_enabled` is `false`.
> 
> ~> **Note:** Not supported to update `public_ip_enabled` from `true` to `false` when `subnet_id` is null or empty

Reviewer: We can consolidate these 
```suggestion
~> **Note:** `public_ip_enabled` cannot be updated from `true` to `false` when `subnet_id` is not set. `public_ip_enabled` must be set to `true` if `subnet_id` is not set and when `public_network_access_enabled` is `false`.
```
